runtime_builders/builder_util.py

```diff
@@ -57,10 +63,57 @@ def get_file_from_gcs(gcs_file, temp_file):
         return False
 
 
-def write_manifest_file(manifest):
-    manifest_contents = yaml.round_trip_dump(manifest,
-                                             default_flow_style=False)
-    write_to_gcs(MANIFEST_FILE, SCHEMA_LINE + manifest_contents)
+def verify_manifest(manifest):
+    """Verify that the provided runtime manifest is valid before publishing.
+
+    Aliases are provided for runtime 'names' that can be included in users'
+    application configuration files: this method ensures that all the aliases
+    can resolve to actual builder files.
+
+    All builders and aliases are turned into nodes in a graph, which is then
+    traversed to be sure that all nodes lead down to a builder node.
+
+    Example formatting of the manifest, showing both an 'alias' and
+    an actual builder file:
+
+    runtimes:
+      java:
+        target:
+          runtime: java-openjdk
+      java-openjdk:
+        target:
+          file: gs://runtimes/java-openjdk-1234.yaml
+    """
+    try:
+        node_graph = {}
+        for key, val in manifest.get('runtimes').iteritems():
+            if key == 'schema_version':
+                continue
+            target = val['target']
```

Comment:
This will crash if there is no target.  Technically this is valid if we remove a runtime.  We would potentially want to leave it in the manifest because we could keep a deprecation message there.  In the client it would print the message, but fail because the runtime no longer exists.